File: order/src/main/java/com/apapedia/order/model/OrderItem.java
package com.apapedia.order.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.Getter;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@JsonIgnoreProperties(value = {"order"}, allowSetters = true)
@Table(name = "order_item")

public class OrderItem {
    @Id
    private UUID orderItemId = UUID.randomUUID();

    @ManyToOne
    @JoinColumn(name = "order_id", referencedColumnName = "orderId")
    private Order order; 

    @NotNull
    @Column(name = "product_id", nullable = false)
    private UUID productId;

    @NotNull
    @Column(name = "quantity", nullable = false)
    private int quantity;

    @NotNull
    @Column(name = "product_name", nullable = false)
    private int productName;

    @NotNull
    @Column(name = "product_price", nullable = false)
    private int productPrice;
}

File: mobileapp/lib/app/data/models/order_item.dart
class OrderItem {
  final String id;
  final String productId;
  final String sellerId;
  final String productName;
  final int productPrice;
  final int quantity;

  OrderItem({
    required this.id,
    required this.productId,
    required this.sellerId,
    required this.productName,
    required this.productPrice,
    required this.quantity,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      id: json['orderItemId'],
      productId: json['productId'],
      sellerId: json['sellerId'],
      productName: json['productName'],
      productPrice: json['productPrice'],
      quantity: json['quantity'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'productId': productId,
        'sellerId': sellerId,
        'productName': productName,
        'productPrice': productPrice,
        'quantity': quantity,
      };
}
